packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/DanglingRepresentationDeletionService.java

```diff
@@ -57,9 +64,17 @@ public boolean isDangling(IEditingContext editingContext, IRepresentation repres
     @Override
     @Transactional
     public void deleteDanglingRepresentations(ICause cause, IEditingContext editingContext) {
-        this.representationMetadataSearchService.findAllMetadataByProject(AggregateReference.to(editingContext.getId())).stream()
-                .filter(representationMetadata -> this.objectSearchService.getObject(editingContext, representationMetadata.getTargetObjectId()).isEmpty())
-                .map(RepresentationMetadata::getId)
-                .forEach(representationId -> this.representationMetadataDeletionService.delete(cause, representationId));
+        var optionalSemanticDataId = new UUIDParser().parse(editingContext.getId())
+                .flatMap(semanticDataId -> this.projectSemanticDataSearchService.findBySemanticDataId(AggregateReference.to(semanticDataId)))
+                .map(ProjectSemanticData::getSemanticData)
+                .map(AggregateReference::getId)
+                .map(UUID::toString);
+
+        optionalSemanticDataId.ifPresent(semanticDataId ->
+                this.representationMetadataSearchService.findAllMetadataByProject(AggregateReference.to(semanticDataId)).stream()
```

Comment:
This is the identifier of the semantic data, not the identifier of the project... Can this work? My comment above may not be viable if we don't scope representations by semantic data id which will not occur right now